### ISSUE TITLE
Problem with ipv6

### ISSUE THREAD

**Issue by kaibox-git** [kataras/iris#1827]
A new project was started with Gin. It works fine. Now I am trying to rewrite it to Iris.
In version 12.1.8 (stable, I thought) I could not find many documented functions like:
app.UseRouter()
ctx.StopWithJSON()
and so on...

I download 12.2.0-alpha4
The project starts, but if I go on any route the panic fires:
`2021/11/24 12:49:59 http: panic serving [::1]:42968: runtime error: invalid memory address or nil pointer dereference
goroutine 66 [running]:
net/http.(*conn).serve.func1()
        /home/kai/.gvm/gos/go1.17.3/src/net/http/server.go:1801 +0xb9
panic({0xd68720, 0x16c23c0})
        /home/kai/.gvm/gos/go1.17.3/src/runtime/panic.go:1047 +0x266
github.com/kataras/iris/v12/core/router.(*Router).ServeHTTP(0x0, {0x1002418, 0xc000632000}, 0x10000c0005ceaf0)
        /home/kai/go/pkg/mod/github.com/kataras/iris/v12@v12.2.0-alpha4/core/router/router.go:319 +0x22
net/http.serverHandler.ServeHTTP({0xc0002f8fc0}, {0x1002418, 0xc000632000}, 0xc0000d2000)
        /home/kai/.gvm/gos/go1.17.3/src/net/http/server.go:2878 +0x43b
net/http.(*conn).serve(0xc0003dc0a0, {0x100a460, 0xc000602420})
        /home/kai/.gvm/gos/go1.17.3/src/net/http/server.go:1929 +0xb08
created by net/http.(*Server).Serve
        /home/kai/.gvm/gos/go1.17.3/src/net/http/server.go:3033 +0x4e8
`
I think its because of ipv6 on my computer (Ubuntu 20.04). All programs detect remote IP as ipv6 since sometime. May be iris.router fails with this? My localhost has IP "::1", not "127.0.0.1". There is no errors in panic list concerning my code and the same handlers work with Gin. Please, help.

**Comment by kaibox-git** [comment]
No, different problem. I started on another server and the same issue with ipv4:

2021/11/24 13:44:20 http: panic serving 10.80.5.122:34390: runtime error: invalid memory address or nil pointer dereference
goroutine 26 [running]:
net/http.(*conn).serve.func1()
	/home/kai/.gvm/gos/go1.17.3/src/net/http/server.go:1801 +0xb9
panic({0xd68720, 0x16c23c0})
	/home/kai/.gvm/gos/go1.17.3/src/runtime/panic.go:1047 +0x266
github.com/kataras/iris/v12/core/router.(*Router).ServeHTTP(0xc0000a21a0, {0x1002418, 0xc0000f6620}, 0xc000266430)
	/home/kai/go/pkg/mod/github.com/kataras/iris/v12@v12.2.0-alpha4/core/router/router.go:319 +0x22
net/http.serverHandler.ServeHTTP({0x1000c60}, {0x1002418, 0xc0000f6620}, 0xc0001f6400)
	/home/kai/.gvm/gos/go1.17.3/src/net/http/server.go:2878 +0x43b
net/http.(*conn).serve(0xc0002777c0, {0x100a460, 0xc0000337d0})
	/home/kai/.gvm/gos/go1.17.3/src/net/http/server.go:1929 +0xb08
created by net/http.(*Server).Serve
	/home/kai/.gvm/gos/go1.17.3/src/net/http/server.go:3033 +0x4e8

**Comment by kaibox-git** [comment]
Found: I missed to change server start from standart:
srv.ListenAndServe()
to specific:
app.Run(iris.Raw(srv.ListenAndServe))

**Comment by kaibox-git** [comment]
It may be interesting for someone. I rewrote the http handlers to iris. Now the binary works on server on port 8080, and I installed gin on port 8081. Considering that the binaries do not differ in the handlers themselves, I carried out load testing with the program https://github.com/codesenberg/bombardier
I did two tests for each port:
IRIS:
./bombardier -c 125 -n 10000000 http://localhost:8080/api/login
Reqs/sec 10743.24
Reqs/sec 10647.30
GIN:
./bombardier -c 125 -n 10000000 http://localhost:8081/api/login
Reqs/sec 10451.21
Reqs/sec 10298.75

Yes, you are right, iris is faster.
